tests/test_conservation.py

```diff
@@ -0,0 +1,377 @@
+import climt
+from sympl import (
+    get_constant, AdamsBashforth,
+    TendencyComponent, TendencyStepper)
+from datetime import timedelta
+import numpy as np
+
+
+class Conservation(object):
+
+    def get_model_state(self, component):
+
+        state = climt.get_default_state([component])
+        return self.modify_state(state)
+
+    def modify_state(self, state):
+        return state
+
+    def get_component_instance(self):
+        pass
+
+    def get_quantity_forcing(self, state):
+        pass
+
+    def get_quantity_amount(self, state):
+        pass
+
+    def get_steppable_component(self):
+
+        component = self.get_component_instance()
+
+        if isinstance(component, TendencyComponent):
+            return AdamsBashforth(component)
+        else:
+            return component
+
+    def get_new_state_and_diagnostics(self, state,
+                                      component, time_step):
+
+        if isinstance(component, TendencyStepper):
+            diag, state = component(state, time_step)
+            state.update(diag)
+
+            return state
+        else:
+            diag, new_state = component(state, time_step)
+            state.update(new_state)
+            state.update(diag)
+
+            return state
+
+    def test_quantity_is_conserved(self):
+
+        component = self.get_steppable_component()
+        state = self.get_model_state(component)
+        time_step = timedelta(seconds=1)
+
+        first_state = self.get_new_state_and_diagnostics(state,
```

Comment:
Not super-necessary for this merge, but generally this is more readable if you put `state` and all that follows on a new line with 4 spaces of indentation relative to the previous line. Such as:
```
first_state = self.get_new_state_and_diagnostics(
    state, component, time_step)
```